pmd-core/src/main/java/net/sourceforge/pmd/properties/GenericMultiValuePropertyDescriptor.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.properties;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import net.sourceforge.pmd.properties.constraints.PropertyConstraint;
+
+
+/**
+ * If we implement schema changes to properties, delimiter logic will probably be scrapped,
+ * and hence the divide between multi-value and single-value property descriptors. We can then
+ * use a single class for all property descriptors.
+ *
+ * @author Clément Fournier
+ * @since 6.10.0
+ */
+final class GenericMultiValuePropertyDescriptor<V, C extends Collection<V>> extends AbstractMultiValueProperty<V> {
+
+
+    private final Set<PropertyConstraint<? super C>> listValidators;
+    private final ValueParser<V> parser;
+    private final Class<V> type;
+
+
+    GenericMultiValuePropertyDescriptor(String name, String description, float uiOrder,
+                                        Collection<V> defaultValue,
+                                        Set<PropertyConstraint<? super C>> listValidators,
+                                        ValueParser<V> parser,
+                                        char delim,
+                                        Class<V> type) {
+        // this cast is safe until 7.0.0
+        super(name, description, (List<V>) defaultValue, uiOrder, delim, false);
+        this.listValidators = listValidators;
+        this.parser = parser;
+        this.type = type;
+
+        String dftValueError = errorFor(new ArrayList<>(defaultValue));
```

Comment:
do we really need the `new ArrayList<>`? can't this be `errorFor(Collections.singletonList(defaultValue))` and minimize allocations?

Reply:
`defaultValue` is a `Collection<V>` but the `errorFor` method (which is in one abstract class and must be implemented that way) takes a `List<V>`. Changing that would impact the countless other multivalue descriptor classes which are deprecated. 

So, yes it's sloppy, but it's transitional and we can't change it until we remove all those deprecated classes.

Reply:
Ah, I think I get what you mean. `Collections.singletonList(defaultValue)` would yield a `List<Collection<V>>` where a `List<V>` is expected. We could however cast `defaultValue` to a `List<V>`, which is safe during the transition and is already done 2 lines above ^^

Reply:
ohh, you are right! thanks!